### ISSUE TITLE
Incorrect url when deploying docker container to webapp in Azure China

### ISSUE THREAD

**Issue by tobols** [Azure/azure-cli#26214]
**Related command**
az webapp config container set

**Describe the bug**
When deploying a docker container to Azure China, the management host url is missing a trailing "/", resulting in an incorrect url.
Full command is:
az webapp config container set --name <WEBAPPNAME> --slot <SLOTNAME> --resource-group <RESOURCEGROUP> --docker-custom-image-name <DOCKERIMAGE> --docker-registry-server-url <DOCKERSERVER> --docker-registry-server-user <USER> --docker-registry-server-password <PASSWORD>

Error message:
The command failed with an unexpected error. Here is the traceback:
HTTPSConnectionPool(host='management.chinacloudapi.cnsubscriptions', port=443): Max retries exceeded with url: /<REMOVED>/resourceGroups/<REMOVED>/providers/Microsoft.Web/sites/<REMOVED>?api-version=2022-03-01 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0607C718>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/connection.py", line 174, in _new_conn
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/util/connection.py", line 73, in create_connection
  File "socket.py", line 955, in getaddrinfo
socket.gaierror: [Errno 11001] getaddrinfo failed

Problem seems to be that there is a missing "/" in: management.chinacloudapi.cnsubscriptions
It should be: management.chinacloudapi.cn/subscriptions

**To Reproduce**
Deploy a docker container with the following command:
az webapp config container set --name <WEBAPPNAME> --slot <SLOTNAME> --resource-group <RESOURCEGROUP> --docker-custom-image-name <DOCKERIMAGE> --docker-registry-server-url <DOCKERSERVER> --docker-registry-server-user <USER> --docker-registry-server-password <PASSWORD>

**Expected behavior**
The container should be successfully deployed to the webapp

**Environment summary**
azure-cli                         2.47.0
core                              2.47.0
telemetry                          1.0.8
Dependencies:
msal                              1.20.0
azure-mgmt-resource               22.0.0

**Additional context**
This problem began after upgrading the azure-cli to version 2.47.0

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by ArtistLL123** [comment]
@yonzhan Is there any update on this issue?

**Comment by yonzhan** [comment]
webapp related CLI commands are owned by service team.

**Comment by tobols** [comment]
@yonzhan if this isn't the correct place to open an issue for webapp related CLI commands, would you instead direct me to the appropriate place please?

**Comment by yosefbs** [comment]
same problem when try to upload zip file via 
`az functionapp deployment source config-zip`

**Comment by ArtistLL123** [comment]
@navba-MSFT Is there any update on this issue?

**Comment by krzyszkowski** [comment]
Hi, any updates on this? We are having same problem deploying to China, as an example below task is failing for AzureChinaCloud subscription

```yaml
- task: AzureCLI@2
  displayName: Task
  inputs:
    azureSubscription: ${{ variables.azureSubscription }}
    scriptType: pscore
    scriptLocation: inlineScript
    inlineScript: |
      az functionapp config appsettings set --name ${{ variables.appName }} --resource-group ${{ variables.rgName }} --setting ANY_SETTING=ANY_VALUE

**Comment by tobols** [comment]
@krzyszkowski @neilboyd great find, thank you!

**Comment by seligj95** [comment]
fixed by: #26921

**Comment by seligj95** [comment]
Closing. This should be fixed in the latest CLI release version 2.51.0. Please update your CLI version and reopen this issue if the problem persists.
